api/proto/teleport/auditlog/v1/auditlog.proto

```diff
@@ -0,0 +1,95 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package teleport.auditlog.v1;
+
+import "google/protobuf/struct.proto";
+import "google/protobuf/timestamp.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/auditlog/v1;auditlogv1";
+
+// AuditLogService provides methods to access audit log.
+service AuditLogService {
+  // StreamUnstructuredSessionEvents streams audit events from a given session recording in an unstructured format.
+  // This endpoint is used by the event handler to retrieve the session events as JSON.
+  rpc StreamUnstructuredSessionEvents(StreamUnstructuredSessionEventsRequest) returns (stream EventUnstructured);
+  // GetUnstructuredEvents gets events from the audit log in an unstructured format.
+  // This endpoint is used by the event handler to retrieve the events as JSON.
+  rpc GetUnstructuredEvents(GetUnstructuredEventsRequest) returns (EventsUnstructured);
+}
+
+// StreamUnstructuredSessionEventsRequest is a request containing needed data to fetch a session recording.
+message StreamUnstructuredSessionEventsRequest {
+  // session_id is the ID for a given session in an UUIDv4 format.
+  string session_id = 1;
+  // StartIndex is the index of the event to resume the stream after.
+  // A start_index of 0 creates a new stream.
+  int32 start_index = 2;
+}
+
+// Order specifies any ordering of some objects as returned in regards to some aspect
+// of said objects which may be trivially ordered such as a timestamp.
+enum Order {
+  ORDER_DESCENDING_UNSPECIFIED = 0;
+  ORDER_ASCENDING = 1;
+}
+
+// GetUnstructuredEventsRequest is a request with the needed data to fetch events.
+message GetUnstructuredEventsRequest {
+  // namespace, if not set, defaults to 'default'
```

Comment:
Nit but the comments for fields of `GetUnstructuredEventsRequest` should probably end with a dot as well, just to keep it consistent across comments in this file. ;)